### PR TITLE
[WIP]rooms_my_tweaks

### PR DESCRIPTION
**Pull request by Nevrdid**
new UpdateAmount role method & handleAttack in myRoom execute exported

### CODE REVIEW

**Comment by democratic-collaboration** [comment]
[democratic collaboration](https://github.com/TooAngel/democratic-collaboration)
Approved reviews will speed up the merge, request changes will slow it down.

Please review the PR to help.

Summoning some reviewers:
 - @TooAngel: I just want to get this merged ASAP
 - @Somotaw: Who are you?
 - @WhiteHalmos: Please look at the code

**Comment by democratic-collaboration** [comment]
DCBOT: Adding WIP label, the title is prefixed with [WIP]

File: src/prototype_room_my.js [hunk 1]
```diff
@@ -397,6 +339,66 @@ Room.prototype.executeRoom = function() {
...
+  this.memory.attackTimer = this.memory.attackTimer || 0;
+
+  var spawns = this.find(FIND_MY_STRUCTURES, {
+    filter: function(object) {
+      return object.structureType === STRUCTURE_SPAWN;
+    }
+  });
+
+  var hostiles = this.find(FIND_HOSTILE_CREEPS, {
+    filter: this.findAttackCreeps
```
Comment (invisioblack) [inline src/prototype_room_my.js]: This does not honor the friends - maybe that needs to be added?

Comment (Nevrdid) [inline src/prototype_room_my.js]: hum dunno, don't have friends xD but i didn't edit that... A good way for always filter them could be to do a general filter method in each room for drop different structure / creeps in differents array as we no more need to filter them later. i'll think about

Comment (invisioblack) [inline src/prototype_room_my.js]: use this function in conjunction?  'brain.isFriend(object.owner.username);'

Comment (invisioblack) [inline src/prototype_room_my.js]: use this function in conjunction?  'brain.isFriend(object.owner.username);'

Comment (Nevrdid) [inline src/prototype_room_my.js]: hum yeah, looks like you just have to include it in afindAttackCreeps in proto_room_defense but it's not for this PR :) Feel free to create it or i'll do it later if i don't forget ^^'

**Pull request closed without merging**